service/src/test/java/bio/terra/lz/futureservice/pact/provider/LandingZoneServicePactProviderTest.java

```diff
@@ -0,0 +1,266 @@
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.web.servlet.MockMvc;
+
+/**
+ * To run these tests against a local set of pacts, comment out the @PactBroker annotation and
+ * replace with an @PactFolder annotation with a path to your pacts
+ */
+@AutoConfigureMockMvc
+@Tag("pact-provider-test")
+@Provider("terra-landing-zone-service")
+@PactBroker
+@SpringBootTest(
+    properties = {
+      "otel.instrumentation.spring-webmvc.enabled=false",
+      "landingzone.landingzone-database.initialize-on-start=false",
+      "landingzone.landingzone-database.upgrade-on-start=false",
+    },
+    classes = LandingZoneApplication.class)
+public class LandingZoneServicePactProviderTest extends BaseSpringUnitTest {
+  private static final String CONSUMER_BRANCH = System.getenv("CONSUMER_BRANCH");
+
+  @MockBean BearerTokenFactory bearerTokenFactory;
+  @MockBean LandingZoneAppService landingZoneAppService;
+  @MockBean LandingZoneJobService landingZoneJobService;
+  @Autowired private MockMvc mockMvc;
+
+  @PactBrokerConsumerVersionSelectors
+  public static SelectorBuilder consumerVersionSelectors() {
+    // The following match condition basically says
+    // If verification is triggered by Pact Broker webhook due to consumer pact change, verify only
+    // the changed pact.
+    // Otherwise, this is a PR, verify all consumer pacts in Pact Broker marked with a deployment
+    // tag (e.g. dev, alpha).
+    if (StringUtils.isBlank(CONSUMER_BRANCH)) {
+      return new SelectorBuilder().mainBranch().deployedOrReleased();
+    } else {
+      return new SelectorBuilder().branch(CONSUMER_BRANCH);
+    }
+  }
+
+  @BeforeEach
+  void setUp(PactVerificationContext context) {
+    context.setTarget(new MockMvcTestTarget(mockMvc));
+
+    when(landingZoneAppService.listAzureLandingZonesDefinitions(any()))
+        .thenReturn(new ApiAzureLandingZoneDefinitionList());
+    when(landingZoneAppService.listAzureLandingZones(any(), any()))
+        .thenReturn(
+            new ApiAzureLandingZoneList()
+                .landingzones(
+                    List.of(
+                        new ApiAzureLandingZone()
+                            .landingZoneId(UUID.randomUUID())
+                            .billingProfileId(UUID.randomUUID())
+                            .version("v1")
+                            .definition("definition")
+                            .region("eastus")
+                            .createdDate(Instant.now().atOffset(ZoneOffset.UTC)))));
+    when(landingZoneAppService.createAzureLandingZone(any(), any(), any()))
```

Comment:
It might eventually make sense to move this mocking to the landing zone service, to capture validations happening there; but I think that's a future improvement, not an immediate need.

It might also still be a nice to have this somewhere where we can make better use of provider state variables and such, but I think that's also something we can revisit when it seems useful.